syntax = "proto3";
package proto_package;

message Empty{
    
}

message Request {
    string user_name = 1;
    string user_message = 2;
}

message Response {
    string user_name = 1;
    string user_response = 2;
}

message Connect_2{
    int32 user_id = 1;
    string user_name = 2;
}

message Connect_1{
    string user_name = 1;
}



service Server_Service {
    rpc Send_Message_to_all (Connect_1) returns (stream Response);
    rpc Connect (Connect_2) returns (stream Response);
    rpc Disconnect (Connect_2) returns (Response);
    rpc Send_message (Request) returns (Empty);
}
